content/chainguard/administration/cloudevents/image-copy-gcr/index.md

```diff
@@ -0,0 +1,196 @@
+weight: 15
+toc: true
+---
+
+Certain interactions with Chainguard resources will emit [CloudEvents](/chainguard/administration/cloudevents/events-reference/) that you or an application can subscribe to. This allows you to do things like receive alerts when a user downloads one or more of your organization's private Images or when a new Image gets added to your organization's registry.
+
+This tutorial is meant to serve as a companion to the [Image Copy GCP](https://github.com/chainguard-dev/platform-examples/tree/main/image-copy-gcp) example application. It will guide you through setting up infrastructure to listen for `push` events on an organization's private registry and mirror any new Images in the registry to a repository in a GCP Artifact Registry repository.
+
+
+## Prerequisites
+
+To follow along with this guide, it is assumed that you have the following set up and ready to use.
+
+* A [verified Chainguard organization](/chainguard/administration/iam-organizations/verified-orgs/) with a private [Registry](/chainguard/chainguard-registry/overview/) and access to [Production Images](/chainguard/chainguard-images/overview/#production-and-developer-images).
+* `chainctl`, the Chainguard command-line interface. You can install this by following our guide on [How to Install `chainctl`](/chainguard/administration/how-to-install-chainctl/).
+* [`terraform`](https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli) to configure a Google Cloud service account, IAM permissions, and deploy the Cloud Run service.
+* A Google Cloud account with a project running. The example application assumes that your project has the following APIs enabled:
+    * [Artifact Registry API](https://cloud.google.com/artifact-registry/docs/reference/rest)
+    * [Cloud Run Admin API](https://cloud.google.com/run/docs/reference/rest)
+* The [`gcloud` CLI](https://cloud.google.com/sdk/docs/install) installed on your local machine. You'll need this to authenticate to Google Cloud over the command line.
+
+
+## Setting up the Terraform configuration
+
+You can find all the code associated with this sample application in our [Platform Examples](https://github.com/chainguard-dev/platform-examples/tree/main/image-copy-gcp) repository on GitHub. 
+
+To set up the sample application, you can create a Terraform configuration file and apply it to set up the necessary resources. To begin, create a new directory to hold the configuration and navigate into it.
+
+```sh
+mkdir ~/gcp-example && cd $_
+```
+
+After navigating into the new directory you can begin creating a Terraform configuration named `main.tf`.
+
+This configuration will consist of a single module. For the purposes of this example, we will call it `image-copy`. This module's `source` value will be the `iac` folder from the application code in the examples repository.
+
+```
+module "image-copy" {
+  source = "github.com/chainguard-dev/platform-examples//image-copy-gcp/iac"
```

Comment:
is the `//` in `platform-examples//image-copy-gcp` correct? afaik this wouldn't trigger errors on Linux (it's ignored) but can cause confusion

Reply:
This is what the example shows in the examples repo: 

but you're right! it does seem like an error. I'll fix it in my doc and see if I can fix it upstream too.